core/src/test/java/tech/tablesaw/table/TableTransposeTest.java

```diff
@@ -0,0 +1,193 @@
+package tech.tablesaw.table;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static tech.tablesaw.TableAssertions.assertTableContents;
+
+import org.junit.jupiter.api.Test;
+import tech.tablesaw.api.BooleanColumn;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.FloatColumn;
+import tech.tablesaw.api.IntColumn;
+import tech.tablesaw.api.LongColumn;
+import tech.tablesaw.api.StringColumn;
+import tech.tablesaw.api.Table;
+
+public class TableTransposeTest {
+
+  @Test
+  void transposeDoubles() {
+    StringColumn label = StringColumn.create("label").append("row1").append("row2").append("row3");
+    DoubleColumn value = DoubleColumn.create("value1").append(1.0).append(1.1).append(1.2);
+    DoubleColumn value2 = DoubleColumn.create("value2").append(2.0).append(2.1).append(2.2);
+
+    Table testTable = Table.create("Data");
+    testTable.addColumns(label);
+    testTable.addColumns(value);
+    testTable.addColumns(value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
```

Comment:
This is a little hard to read to me because it's essentially transposed. What if we used `assertTablesEqual` instead? (see comment above regarding existence of that method)

Reply:
I looked at assertTablesEqual before I implemented this, but it requires the developer to construct a Table for the expected result. That is a lot of extra boilerplate that I would like to avoid. I find this to be much clearer and quicker to use. 
Please note, we are testing transpose, so the results look transposed because they are, rather than because assertTableContents is expecting input in an illogical transposed format. For another feature I think it would look less confusing.

Reply:
Using `assertTablesEqual` is actually fewer lines of code because you no longer need to separately check the table name:

```
    assertTablesEquals(
        Table.create(testTable.name()
            DoubleColumn.create("value1", new double[] {1.0, 1.1, 1.2}),
            DoubleColumn.create("value2", new double[] {2.0, 2.1, 2.2})
        ),
        result);
```

Reply:
It is more boilerplate though - more tokens, doing all the additional Column.create()/Table.create()'s drove me crazy. The version without that is much easier to eyeball as you scan through the test, as it looks much more like a CSV. I find it is more intuitive to see the headers in a row at the top, as they would be if the table was displayed.
More fundamentally, in my view that is a more brittle test, as it is using the class we are trying to test (Table) for the expected values, and if the internals of the class break we are more susceptible to the test still telling us everything is fine mistakenly. For instance, I wouldn't construct an ArrayList of expected values to check the contents of an ArrayList.